tokens/usecase/tokens_usecase.go

```diff
@@ -349,26 +391,69 @@ func (t *tokensUseCase) getPricesForBaseDenom(ctx context.Context, baseDenom str
 }
 
 func (t *tokensUseCase) getChainScalingFactorMut(precision int) (osmomath.Dec, bool) {
-	result, ok := t.precisionScalingFactorMap[precision]
-	return result, ok
+	result, ok := t.precisionScalingFactorMap.Load(precision)
+	if !ok {
+		return osmomath.Dec{}, false
+	}
+
+	v, ok := result.(osmomath.Dec)
+	if !ok {
+		return osmomath.Dec{}, false
+	}
+
+	return v, true
+}
+
+// UpdateAssetsAtHeightInterval updates assets at configured height interval.
+// Internally, it calls LoadTokensFromChainRegistry as a goroutine.
+func (t *tokensUseCase) UpdateAssetsAtHeightInterval(height uint64) {
```

Comment:
nit: I think having this specified for the caller would be helpful
```suggestion
func (t *tokensUseCase) UpdateAssetsAtHeightIntervalAsync(height uint64) {
```

Reply:
Thanks, it's updated including in all other relevant places.